### PR TITLE
add: text pieces and other annotations

### PR DESCRIPTION
**Pull request by sayakpaul**

### CODE REVIEW

File: multi_label_trainer_tfidf.ipynb [hunk 1]
```diff
@@ -4,10 +4,9 @@
   "metadata": {
```
Comment (soumik12345) [inline multi_label_trainer_tfidf.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #8.    import pandas as pd
</pre><p>Also add <code>import matplotlib.pyplot as plt</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/multi-label-text-classification/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (soumik12345) [inline multi_label_trainer_tfidf.ipynb:4]: <p>For an end-to-end demonstration of classification with imbablanced data, refer to <a href="https://keras.io/examples/structured_data/imbalanced_classification/" target="_blank">Imbalanced classification: credit card fraud detection</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/multi-label-text-classification/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (soumik12345) [inline multi_label_trainer_tfidf.ipynb:4]: <pre class="ql-syntax" spellcheck="false">def make_dataset(dataframe, is_train=True):
&nbsp; &nbsp; label_binarized = mlb.transform(dataframe["terms"].values)
&nbsp; &nbsp; dataset = tf.data.Dataset.from_tensor_slices(
&nbsp; &nbsp; &nbsp; &nbsp; (dataframe["summaries"].values, label_binarized)
&nbsp; &nbsp; )
&nbsp; &nbsp; dataset = dataset.shuffle(batch_size * 10) if is_train else dataset
&nbsp; &nbsp; dataset = dataset.map(unify_text_length).cache()
&nbsp; &nbsp; return dataset.batch(batch_size).prefetch(tf.data.AUTOTUNE)
</pre>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/multi-label-text-classification/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (soumik12345) [inline multi_label_trainer_tfidf.ipynb:4]: <pre class="ql-syntax" spellcheck="false">train_dataset = make_dataset(train_df, is_train=True)
validation_dataset = make_dataset(val_df, is_train=False)
test_dataset = make_dataset(test_df, is_train=False)
</pre><p> </p><p>I think keyword arguments make the code more readable.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/multi-label-text-classification/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (soumik12345) [inline multi_label_trainer_tfidf.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #14.                keras.Input(shape=(), dtype=tf.string),
</pre><p>Do we need to add <code>keras.Input</code>  in a <code>keras.Sequential</code>  model?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/multi-label-text-classification/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (soumik12345) [inline multi_label_trainer_tfidf.ipynb:4]: <pre class="ql-syntax" spellcheck="false">history = shallow_mlp_model.fit(train_dataset, validation_data=validation_dataset, epochs=epochs)

def plot_result(item):
    plt.plot(history.history[item], label=item)
    plt.plot(history.history["val_" + item], label="val_" + item)
    plt.xlabel("Epochs")
    plt.ylabel(item)
    plt.title("Train and Validation {} Over Epochs".format(item), fontsize=14)
    plt.legend()
    plt.grid()
    plt.show()


plot_result("loss")
plot_result("categorical_accuracy")
</pre><p> </p><p>We should add plots for the training history.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/multi-label-text-classification/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (soumik12345) [inline multi_label_trainer_tfidf.ipynb:4]: <p>Let's add some sample inference post evaluation. If the results are not up to the mark, we can add suggestions as to how to get better results in the conclusion section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/soumik12345/multi-label-text-classification/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

**Comment by soumik12345** [comment]
Hi @sayakpaul , I added my reviews.

**Comment by sayakpaul** [comment]
@soumik12345 feel free to incorporate these suggestions in a PR and we can take it from there.

**Comment by sayakpaul** [comment]
@soumik12345 two things:

* Add one/two lines noting your observations after the training plot. 
* Add a few inference results just as you had mentioned.

**Comment by soumik12345** [comment]
> @soumik12345 two things:
> 
> * Add one/two lines noting your observations after the training plot.
> * Add a few inference results just as you had mentioned.

Raising new PR for inference.
